llvm/include/llvm/Analysis/MemoryBuiltins.h

```diff
@@ -215,7 +216,8 @@ template <typename T, class C> struct SizeOffsetType {
 /// \p APInts.
 struct SizeOffsetAPInt : public SizeOffsetType<APInt, SizeOffsetAPInt> {
   SizeOffsetAPInt() = default;
-  SizeOffsetAPInt(APInt Size, APInt Offset) : SizeOffsetType(Size, Offset) {}
+  SizeOffsetAPInt(APInt Size, APInt Offset)
+      : SizeOffsetType(std::move(Size), std::move(Offset)) {}
 
   static bool known(APInt V) { return V.getBitWidth() > 1; }
```

Comment:
While you're at it - this should pass by const ref?

Reply:
https://github.com/dtcxzyw/llvm-project/commit/32d6611af69bf4e76373f9bc7d9649650f760e48
I will file a separate PR later.